gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/VoltageCheckService.java

```diff
@@ -42,6 +42,10 @@ public VoltageCheckService(Logger businessLogger, UrlValidationService urlValida
     public Optional<VoltageMonitoringResult> runVoltageCheck(SweData sweData, DichotomyResult<SweDichotomyValidationData> dichotomyResult, DichotomyDirection direction) {
 
         if ((direction != DichotomyDirection.ES_FR && direction != DichotomyDirection.FR_ES) || !dichotomyResult.hasValidStep()) {
+
```

Comment:
It's better to separate the two conditions in the first if :
 if ((direction != DichotomyDirection.ES_FR && direction != DichotomyDirection.FR_ES) {
if (!dichotomyResult.hasValidStep()) {
businessLogger.warn("Voltage check is in failure because no valid step is available to run it.");
} else {
 businessLogger.info("Running voltage check");
....}

Reply:
not possible